modules/calib3d/src/usac/estimator.cpp

```diff
@@ -590,6 +590,66 @@ ReprojectionErrorAffine::create(const Mat &points) {
     return makePtr<ReprojectionDistanceAffineImpl>(points);
 }
 
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Computes forward reprojection error for affine 3D transformation.
+class ReprojectionDistanceAffine3DImpl : public ReprojectionErrorAffine3D {
+private:
+    /*
+     * m11 m12 m13 m14
+     * m21 m22 m23 m24
+     * m31 m32 m33 m34
+     * 0   0   0   1
+     */
+    const Mat * points_mat;
+    const float * const points;
+    float m11, m12, m13, m14, m21, m22, m23, m24, m31, m32, m33, m34;
```

Comment:
By the way, we have `Matx` and `Vec` structs for that

Reply:
Yes, I prefer to use them, indeed.
But, I just follow [the existing 2D version](https://github.com/opencv/opencv/blob/c63d79c5b16fcbbec46f1b8bb871dab2274e2b01/modules/calib3d/src/usac/estimator.cpp#L539) in favor.

> class ReprojectionDistanceAffineImpl : public ReprojectionErrorAffine {
> private:
>     const Mat * points_mat;
>     const float * const points;
>     float m11, m12, m13, m21, m22, m23;
>